Failing alert build after doLittle merge

Build FAILED.


"C:\projects\cbs-q2clx\Source\Alert\Alert.sln" (Build target) (1) ->

"C:\projects\cbs-q2clx\Source\Alert\Policies\Policies.csproj" (default target) (9) ->

(CoreCompile target) -> 

  AlertMessages\AlertFeedbackService.cs(4,7): error CS0246: The type or namespace name 'Infrastructure' could not be found (are you missing a using directive or an assembly reference?) [C:\projects\cbs-q2clx\Source\Alert\Policies\Policies.csproj]

  CaseReportPolicy.cs(6,7): error CS0246: The type or namespace name 'Infrastructure' could not be found (are you missing a using directive or an assembly reference?) [C:\projects\cbs-q2clx\Source\Alert\Policies\Policies.csproj]

  AlertMessages\AlertFeedbackService.cs(14,32): error CS0246: The type or namespace name 'Feature' could not be found (are you missing a using directive or an assembly reference?) [C:\projects\cbs-q2clx\Source\Alert\Policies\Policies.csproj]

  AlertMessages\AlertFeedbackService.cs(18,26): error CS0246: The type or namespace name 'IEventEmitter' could not be found (are you missing a using directive or an assembly reference?) [C:\projects\cbs-q2clx\Source\Alert\Policies\Policies.csproj]

  AlertMessages\AlertFeedbackService.cs(21,107): error CS0246: The type or namespace name 'IEventEmitter' could not be found (are you missing a using directive or an assembly reference?) [C:\projects\cbs-q2clx\Source\Alert\Policies\Policies.csproj]

  CaseReportPolicy.cs(14,37): error CS0246: The type or namespace name 'IEventProcessor' could not be found (are you missing a using directive or an assembly reference?) [C:\projects\cbs-q2clx\Source\Alert\Policies\Policies.csproj]

  CaseReportPolicy.cs(16,32): error CS0246: The type or namespace name 'Feature' could not be found (are you missing a using directive or an assembly reference?) [C:\projects\cbs-q2clx\Source\Alert\Policies\Policies.csproj]

  CaseReportPolicy.cs(19,26): error CS0246: The type or namespace name 'IEventEmitter' could not be found (are you missing a using directive or an assembly reference?) [C:\projects\cbs-q2clx\Source\Alert\Policies\Policies.csproj]

  CaseReportPolicy.cs(24,59): error CS0246: The type or namespace name 'IEventEmitter' could not be found (are you missing a using directive or an assembly reference?) [C:\projects\cbs-q2clx\Source\Alert\Policies\Policies.csproj]


    0 Warning(s)

    9 Error(s)

--- a/Source/Alert/Policies/AlertMessages/AlertFeedbackService.cs
+++ b/Source/Alert/Policies/AlertMessages/AlertFeedbackService.cs
@@ -1,28 +1,28 @@
 using System.Collections.Generic;
 using System.Linq;
 using Events;
-using Infrastructure.Application;
 using doLittle.Events;
 using Read;
 using Read.Disease;
 using Policies.AlertMessages;
+using doLittle.Runtime.Events.Coordination;
+using doLittle.Runtime.Events;
+using doLittle.Runtime.Transactions;
 
 namespace Policies
 {
     public class AlertFeedbackService : IAlertFeedbackService
     {
-        public static readonly Feature Feature = "Alert";
-
         private readonly ISmsSendingService _smsSendingService;
         private readonly IDataCollectors _dataCollectors;
-        private readonly IEventEmitter _eventEmitter;
         private readonly IMessageTemplateService _messageTemplateService;
+        private readonly IUncommittedEventStreamCoordinator _uncommittedEventStreamCoordinator;
 
-        public AlertFeedbackService(ISmsSendingService smsSendingService, IDataCollectors dataCollectors, IEventEmitter eventEmitter, IMessageTemplateService messageTemplateService)
+        public AlertFeedbackService(ISmsSendingService smsSendingService, IDataCollectors dataCollectors, IMessageTemplateService messageTemplateService, IUncommittedEventStreamCoordinator uncommittedEventStreamCoordinator)
         {
             _smsSendingService = smsSendingService;
             _dataCollectors = dataCollectors;
-            _eventEmitter = eventEmitter;
+            _uncommittedEventStreamCoordinator = uncommittedEventStreamCoordinator;
             _messageTemplateService = messageTemplateService;
         }
 
@@ -38,7 +38,7 @@ namespace Policies
             {
                 string text = _messageTemplateService.ComposeMessage(EMessageTemplateNames.AlertRaisedFeedbackToDataVerifiers, healthRisk);
                 _smsSendingService.SendSMS(new [] { dataVerifier.Phone }, "");
-                _eventEmitter.Emit(Feature, new SmsSentEvent() { RecipientName = dataVerifier.Name, RecipientPhoneNumber = dataVerifier.Phone, SmsText = text});
+                ApplyEvent(new SmsSentEvent() { RecipientName = dataVerifier.Name, RecipientPhoneNumber = dataVerifier.Phone, SmsText = text});
             }
 
             // send sms to all data collecors
@@ -46,8 +46,17 @@ namespace Policies
             {
                 string text = _messageTemplateService.ComposeMessage(EMessageTemplateNames.AlertRaisedFeedbackToDataCollectors, healthRisk);
                 _smsSendingService.SendSMS(new[] { dataCollector.Phone }, "");
-                _eventEmitter.Emit(Feature, new SmsSentEvent() { RecipientName = dataCollector.Name, RecipientPhoneNumber = dataCollector.Phone, SmsText = text });
+                ApplyEvent(new SmsSentEvent() { RecipientName = dataCollector.Name, RecipientPhoneNumber = dataCollector.Phone, SmsText = text });
             }
+        }
+
+
+        void ApplyEvent(IEvent @event)
+        {
+            // Todo: Temporary fix - we're not supposed to do this, awaiting the new Policy building block in doLittle to be ready
+            var stream = new UncommittedEventStream(null);
+            stream.Append(@event, EventSourceVersion.Zero.NextCommit());
+            _uncommittedEventStreamCoordinator.Commit(TransactionCorrelationId.New(), stream);
         }
     }
 

--- a/Source/Alert/Policies/CaseReportPolicy.cs
+++ b/Source/Alert/Policies/CaseReportPolicy.cs
@@ -1,30 +1,28 @@
 using System;
-using System.Collections.Generic;
-using System.Linq;
 using Events;
 using Events.External;
-using Infrastructure.Application;
-using doLittle.Events;
 using Read;
 using Read.Alert;
 using Read.HealthRiskObjects;
+using doLittle.Events.Processing;
+using doLittle.Runtime.Events.Coordination;
+using doLittle.Runtime.Transactions;
+using doLittle.Runtime.Events;
 
 namespace Policies
 {
-    public class CaseReportPolicy : IEventProcessor
+    public class CaseReportPolicy : ICanProcessEvents
     {
-        public static readonly Feature Feature = "CaseReport";
-
         private readonly ICaseReports _caseReports;
-        private readonly IEventEmitter _eventEmitter;
         private readonly IHealthRisks _healthRisks;
         private readonly IAlerts _alerts;
         private readonly IAlertFeedbackService _feedbackService;
+        private readonly IUncommittedEventStreamCoordinator _uncommittedEventStreamCoordinator;
 
-        public CaseReportPolicy(ICaseReports caseReports, IEventEmitter eventEmitter, IHealthRisks healthRisks, IAlertFeedbackService feedbackService, IAlerts alerts)
+        public CaseReportPolicy(ICaseReports caseReports, IHealthRisks healthRisks, IAlertFeedbackService feedbackService, IAlerts alerts, IUncommittedEventStreamCoordinator uncommittedEventStreamCoordinator)
         {
+            _uncommittedEventStreamCoordinator = uncommittedEventStreamCoordinator;
             _caseReports = caseReports;
-            _eventEmitter = eventEmitter;
             _healthRisks = healthRisks;
             _feedbackService = feedbackService;
             _alerts = alerts;
@@ -79,13 +77,16 @@ namespace Policies
                 alert.CaseReports.Add(caseReport);
                 _alerts.Save(alert);
 
-                _eventEmitter.Emit(Feature, new AlertRaised
-                {
-                });
+
+                // Todo: Temporary fix - we're not supposed to do this, awaiting the new Policy building block in doLittle to be ready
+                var stream = new UncommittedEventStream(null);
+                stream.Append(new AlertRaised(), EventSourceVersion.Zero.NextCommit());
+                _uncommittedEventStreamCoordinator.Commit(TransactionCorrelationId.New(), stream);
+
                 _feedbackService.SendFeedbackToDataCollecorsAndVerifiers(latestReports);
             }
         }
 
-        
+
     }
 }

--- a/Source/Alert/Policies/Policies.csproj
+++ b/Source/Alert/Policies/Policies.csproj
@@ -10,6 +10,7 @@
 
   <ItemGroup>
     <PackageReference Include="doLittle.Runtime.Events" Version="2.0.0-alpha*"/>
+    <PackageReference Include="doLittle.Runtime.Events.Coordination" Version="2.0.0-alpha*"/>
   </ItemGroup>
  
 

